src/Microsoft.VisualBasic/tests/VB/DriveInfo.Windows.Tests.vb

```diff
@@ -0,0 +1,182 @@
+
+        <Fact>
+        Public Shared Sub TestDriveProperties_()
+            If PlatformDetection.IsInAppContainer Then
+                Exit Sub
+            End If
+            Dim validDrive As IO.DriveInfo = FileSystem.Drives().Where(Function(d) d.DriveType = IO.DriveType.Fixed).First()
+                Dim isReady As Boolean = validDrive.IsReady
+            Assert.NotNull(validDrive.Name)
+            Assert.NotNull(validDrive.RootDirectory.Name)
+            Assert.NotNull(validDrive.DriveFormat)
+            Assert.[True](validDrive.AvailableFreeSpace > 0)
+            Assert.[True](validDrive.TotalFreeSpace > 0)
+            Assert.[True](validDrive.TotalSize > 0)
+            Assert.NotNull(validDrive.VolumeLabel)
+        End Sub
+
+        <Fact>
+        <PlatformSpecific(TestPlatforms.Windows)>
+        <SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Accessing drive format is not permitted inside an AppContainer.")>
+        Public Shared Sub TestDriveFormat()
+            Dim validDrive As IO.DriveInfo = FileSystem.Drives().Where(Function(d) d.DriveType = IO.DriveType.Fixed).First()
+            Const VolNameLen As Integer = 50
+            Dim VolumeName As StringBuilder = New StringBuilder(VolNameLen)
+            Const FileSystemNameLen As Integer = 50
+            Dim FileSystemName As StringBuilder = New StringBuilder(FileSystemNameLen)
+            Dim serialNumber, maxFileNameLen, fileSystemFlags As Integer
+            Dim r As Boolean = GetVolumeInformation(validDrive.Name, VolumeName, VolNameLen, serialNumber, maxFileNameLen, fileSystemFlags, FileSystemName, FileSystemNameLen)
+            Dim FileSystemType As String = FileSystemName.ToString
+
+            If r Then
+                Assert.Equal(FileSystemType, validDrive.DriveFormat)
+            Else
+                Assert.Throws(Of IO.IOException)(Function() validDrive.DriveFormat)
+            End If
+
+        End Sub
+
+        <Fact>
+        <PlatformSpecific(TestPlatforms.Windows)>
+        Public Shared Sub TestDriveType()
+            Dim validDrive As IO.DriveInfo = FileSystem.Drives().Where(Function(d) d.DriveType = IO.DriveType.Fixed).First()
+            Dim expectedDriveType As Integer = GetDriveType(validDrive.Name)
+            Assert.Equal(CType(expectedDriveType, IO.DriveType), validDrive.DriveType)
+        End Sub
+
+        <Fact>
+        <PlatformSpecific(TestPlatforms.Windows)>
+        <SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "GetDiskFreeSpaceEx blocked in AC")>
+        Public Shared Sub TestValidDiskSpaceProperties()
+            Dim win32Result As Boolean
+            Dim fbUser As Long = -1
+            Dim tbUser As Long
+            Dim fbTotal As Long
+            Dim drive As IO.DriveInfo
+            drive = FileSystem.Drives().Where(Function(d) d.DriveType = IO.DriveType.Fixed).First()
+            If drive.IsReady Then
+                win32Result = GetDiskFreeSpaceEx(drive.Name, fbUser, tbUser, fbTotal)
```

Comment:
All this native interop will of course only work on Windows. To finish the job, we need to make it work on Unix, and whereever possible we would want to reuse existing code. For example, the existing `DriveInfo.AvailableFreeSpace` can get this value on all OS without any interop.

If you feel like this change is big enough already, it's fine to disable the tests on Unix for now, if we open an issue to track. But it will need doing ...

Reply:
Oops, these are tests. Same though -- I am sure you don't want to write all this native interop again for Unix. I would just call the System.IO methods directly, if VB is just calling through to those. Tests can assume what System.IO.* produces is correct. Hopefully you can eliminate all native interop in your tests.

Reply:
I think I do, some of this is left over from when I started. Now I am just testing places where VB does something different from System.IO.